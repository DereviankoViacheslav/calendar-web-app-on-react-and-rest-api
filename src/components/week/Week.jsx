import React from 'react';
import Day from '../day';
import moment from 'moment';

function getEventsOfDay(events, date) {
  return events.filter((event) => {
    const endDateEvent = moment(event.endDate, 'YYYY-MM-DD');
    const currentDay = moment(date, 'YYYY-MM-DD');
    return moment(currentDay).isSame(endDateEvent);
  });
}

function Week({ date, listEvents, onShowPopup }) {
  let nextDay = moment(date).format('YYYY-MM-DD');
  const days = [...new Array(7)].map(() => {
    const events = getEventsOfDay(listEvents, nextDay);
    const dayElem = <Day key={nextDay} date={nextDay} events={events} onShowPopup={onShowPopup} />;
    nextDay = moment(nextDay).add('days', 1).format('YYYY-MM-DD');
    return dayElem;
  });

  return (
    <div className="days">
      {days}
    </div>
  );
}

export default Week;