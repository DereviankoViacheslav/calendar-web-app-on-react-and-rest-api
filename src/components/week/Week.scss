.week {
  display: flex;
  width: 100%;
  padding-left: 5px;
  background: repeating-linear-gradient(180deg,
      #dadce0,
      #dadce0 1px,
      #fff 1px,
      #fff 42px);
}