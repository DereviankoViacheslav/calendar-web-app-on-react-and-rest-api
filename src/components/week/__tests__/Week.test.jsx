import React from 'react';
import { shallow } from 'enzyme';
import Week from '../Week'

describe('<Week />', () => {
  it('Get snapshot', () => {
    const props = {
      onShowPopup: jest.fn(),
      listEvents: [],
      date: {},
    }
    const wrappedComponent = shallow(<Week {...props} />);
    expect(wrappedComponent).toMatchSnapshot();
  });
  
  it('should render Week', () => {
    const props = {
      onShowPopup: jest.fn(),
      listEvents: [],
      date: {},
    }
    const wrappedComponent = shallow(<Week {...props} />);
    expect(wrappedComponent.find('.week').exists()).toBeTruthy();
  });
});