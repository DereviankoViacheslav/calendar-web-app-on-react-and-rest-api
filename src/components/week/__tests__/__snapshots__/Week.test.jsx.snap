// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Week /> Get snapshot 1`] = `
<div
  className="week"
>
  <Day
    date="2020-02-15"
    events={Array []}
    key="2020-02-15"
    onShowPopup={[MockFunction]}
  />
  <Day
    date="2020-02-16"
    events={Array []}
    key="2020-02-16"
    onShowPopup={[MockFunction]}
  />
  <Day
    date="2020-02-17"
    events={Array []}
    key="2020-02-17"
    onShowPopup={[MockFunction]}
  />
  <Day
    date="2020-02-18"
    events={Array []}
    key="2020-02-18"
    onShowPopup={[MockFunction]}
  />
  <Day
    date="2020-02-19"
    events={Array []}
    key="2020-02-19"
    onShowPopup={[MockFunction]}
  />
  <Day
    date="2020-02-20"
    events={Array []}
    key="2020-02-20"
    onShowPopup={[MockFunction]}
  />
  <Day
    date="2020-02-21"
    events={Array []}
    key="2020-02-21"
    onShowPopup={[MockFunction]}
  />
</div>
`;
